src/providers/crypto/crypto_provider/model.rs

```diff
@@ -0,0 +1,77 @@
+extern crate ed25519_dalek;
+extern crate hex;
+use crate::providers::crypto::{users::User, Ed25519, Sr25519};
+extern crate rand;
+
+use rand_core::OsRng;
+use schnorrkel::{signing_context, Keypair, PublicKey, Signature};
+
+use ed25519_dalek::ed25519::signature::SignerMut;
+
+impl crate::CryptoProvider for Ed25519 {
+    type Transaction = String;
+    type Signature = String;
+    type Error = String;
```

Comment:
Что-то в духе такого должно здесь быть. И тогда 27 строка будет более правильно.

```suggestion
    type Error = anyhow::Error;
```